src/modules/player/AudioPlayer/AudioPlayer.res

```diff
@@ -1,95 +1,191 @@
+%%raw("import './AudioPlayer.css'")
+
+@scope("JSON") @val external parseIntoMyData: string => Playlist.songs = "parse"
+@module("../../../assets/jesus-dancing.gif") external jesus: string = "default"
 
 module AudioHtmlBindings = {
   @send external pause: Dom.element => unit = "pause"
   @send external play: Dom.element => unit = "play"
+  @set external setCurrentTime: (Dom.element, int) => unit = "currentTime"
   @get external currentTime: Dom.element => int = "currentTime"
   @get external duration: Dom.element => int = "duration"
-  @get external waiting: Dom.element => bool = "waiting"
   @send external addEventListener: (Dom.element, string, unit => unit) => unit = "addEventListener"
-  @send external removeEventListener: (Dom.element, string, unit => unit) => unit = "removeEventListener"
+  @send
+  external removeEventListener: (Dom.element, string, unit => unit) => unit = "removeEventListener"
 }
 
-let getCurrent = (audioRef: React.ref<Js.Nullable.t<Dom.element>>) : option<Dom.element> => {
+let getCurrent = (audioRef: React.ref<Js.Nullable.t<Dom.element>>): option<Dom.element> => {
   audioRef.current->Js.Nullable.toOption
 }
 
+let formatTime = (secs: int): string => {
+  let minutes: int = secs / 60
+  let seconds: int = mod(secs, 60) / 1
+
+  let formattedMinutes = Belt.Int.toString(minutes)
+  let formattedSeconds = Belt.Int.toString(seconds)
+
+  if formattedSeconds->String.length < 2 {
+    formattedMinutes ++ ":0" ++ formattedSeconds
+  } else {
+    formattedMinutes ++ ":" ++ formattedSeconds
+  }
+}
+
+module PlayListSource = {
+  let getSongs = (): array<Playlist.song> => {
+    let playerSongs: Playlist.songs = parseIntoMyData(Songs.songs)
+    playerSongs.songs
+  }
+}
+
+module MysticImage = {
+  @react.component
+  let make = (~loading: bool, ~playing: bool) => {
+    <div>
+      {switch (loading, playing) {
+      | (true, false) => <img src="https://media.tenor.com/jfmI0j5FcpAAAAAM/loading-wtf.gif" />
+      | (false, true) => <img src={jesus} />
+      | _ => React.null
+      }}
+    </div>
+  }
+}
+
 module AudioControls = {
   @react.component
-  let make = (~audioRef: React.ref<Js.Nullable.t<Dom.element>>) => {
+  let make = (~audioRef: React.ref<Js.Nullable.t<Dom.element>>, ~song: Playlist.song) => {
     let (isPlaying, setIsPlaying) = React.useState(_ => false)
```

Comment:
maybe all these different state variables could be grouped in a more meaningful (single) type with different states